bin/embark

```diff
@@ -1,15 +1,1023 @@
+    embarklog[loglevel]('JSON parse', `Failed to parse ${basename} data.`);
+    embarklog[loglevel]('JSON parse', `${basename} must be actual JSON, not just JavaScript.`);
+  }
+}
+
+function reportUnparsable_EmbarkJson(loglevel) {
+  console[loglevel]();
+  embarklog[loglevel]('', `Could not parse your DApp's embark.json file`);
+}
+
+function reportUnparsable_PkgJsonEmbark(kind, loglevel) {
+  console[loglevel]();
+  embarklog[loglevel](
+    `Could not parse ${kind} embark's package.json file`
+  );
+}
+
+function reportUnparsable_PkgJsonLocal(loglevel) {
+  embarklog[loglevel]('', `Could not parse a local package.json file`);
+}
+
+function reportUnsupportedNode(
+  bad, filepath, kind, loglevel, missing, rangeDefault, rangeSupplied, pkg,
+  procVer, range) {
+  blankLineMaybe(loglevel);
+  function report(qual, invalid) {
+    embarklog[loglevel]('file', filepath);
+    embarklog[loglevel](
+      'engine',
+      `package.json of ${kind} ${pkg} does not specify ${qual ? qual : ''}%j`,
+      {engines: {node: '[semver]'}}
+    );
+    if (invalid) {
+      embarklog[loglevel](
+        'engine',
+        `Specified: %j`, {engines: {node: rangeSupplied}}
+      );
+    }
+    embarklog[loglevel](
+      'engine',
+      `Defaulting to: %j`, {engines: {node: rangeDefault}}
+    );
+    console[loglevel]();
+  }
+  if (missing) { report(); }
+  if (bad) { report('a valid ', true); }
+  embarklog[loglevel]('notsup', `Unsupported runtime`);
+  embarklog[loglevel](
+    'notsup',
+    `${kind} ${pkg} is not compatible with your version of node`
+  );
+  embarklog[loglevel]('notsup', `Required:`, range);
+  embarklog[loglevel]('notsup', `Actual:`, procVer);
+}
+
+// -- selectors ----------------------------------------------------------------
+
+function select(invoked, local) {
+  var embark;
+  if (local && local.binrealpath !== invoked.binrealpath) {
+    embark = local;
+  } else {
+    embark = invoked;
+  }
+  return embark;
+}
+
+function selectLocal(containing, installed) {
+  var local;
+  if (containing.binrealpath &&
+      (!installed.binrealpath ||
+       subdir(
+         installed.pkgJsonLocalExpected.dirname,
+         containing.pkgDir
+       ))) {
+    local = containing;
+  }
+  if (installed.binrealpath &&
+      (!containing.binrealpath ||
+       subdir(containing.pkgDir, installed.pkgDir))) {
+    local = installed;
+  }
+  if (local) { local.log(); }
+  return local;
+}
+
+// -- utils --------------------------------------------------------------------
+
+function exitWithError(code) {
+  blankLineTrailingMaybe('error');
+  process.exit(code || 1);
+}
+
+function initCwd() {
+  var initCwd = process.env.INIT_CWD || process.cwd();
+  // allow for env override
+  initCwd = process.env.DAPP_PATH || initCwd;
+  return initCwd;
+}
+
+function isDappCmd(cmd) {
+  return [
+    undefined,
+    '-V',
+    '--version',
+    '-h',
+    '--help',
+    'new',
+    'demo',
+    'version',
+    'help'
+  ].indexOf(cmd) === -1;
```

Comment:
What about other commands like graph/console (there might be other)

Reply:
The commands/options not in the array, including `graph` and `console`, are commands that the shim requires to be run in a DApp context. Does that seem right? I may have misunderstood your comment.

Reply:
Understood, thx for clarification